### PR TITLE
Enhancement in setup repos

### PR DESCRIPTION
**Pull request by Ameenasuhani**
Signed-off-by: Ameenasuhani <ameenasuhani@gmail.com>

# Description
Enhancement in setup repos method to use custom options like Tools,MON,OSD repos

Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-HESZM3/

### CODE REVIEW

**Comment by Ameenasuhani** [comment]
test_cephadm.py logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8C3R0M/Deploy_cluster_using_cephadm_0.log
test_bootstrap.py logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-FC7N75/Cephadm_Bootstrap_with_custom_dashboard_credentials_0.log

File: tests/ceph_installer/test_cephadm.py [hunk 1]
```diff
@@ -130,7 +131,7 @@ def run(ceph_cluster: Ceph, **kwargs) -> int:
             if command == "shell":
                 cephadm.shell(**cfg)
                 continue
-
+            cfg["base_url"] = base_url
```
Comment (psathyan) [inline tests/ceph_installer/test_cephadm.py:134]: I see a potential regression issue for other methods. Would it be possible to perform an E2E run?

Comment (Ameenasuhani) [reply tests/ceph_installer/test_cephadm.py:134]: @psathyan i have added logs in comment 1 http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-8C3R0M/ for test_cephadm.py as well. Is there something else to test E2E?

Comment (psathyan) [reply tests/ceph_installer/test_cephadm.py:134]: As discussed, could we use `self.config['base_url']`?

**Comment by Ameenasuhani** [comment]
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-IPKFXQ/
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-EGUPQF/

**Review by psathyan — APPROVED** [review]

**Pull request merged**